just: 0.9.1 -> 0.9.4

06kellyjac opened the pull request:
<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!
List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://nixos.org/manual/nixpkgs/unstable/#chap-reviewing-contributions
-->

###### Motivation for this change

Bump just to `0.9.4`

Also:
Skip `run_shebang` test as it randomly fails
Add changelog
Add myself as a maintainer, if you don't mind @xrelkd

###### Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- [X] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [X] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [x] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [X] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Added a release notes entry if the change is major or breaking
- [X] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

06kellyjac commented:
Result of `nixpkgs-review pr 125653` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>just</li>
  </ul>
</details>

r-rmcgibbo commented:
Result of `nixpkgs-review pr 125653` at f711b0f4 run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package failed to build:</summary>
  <ul>
    <li><a href="https://gist.github.com/c4abb4cd1293fe0dea89524895c4aefb">just</a></li>
  </ul>
</details>

Note that build failures [may predate](https://github.com/nix-community/hydra-check) this PR, and could be nondeterministic or hardware dependent.
Please exercise your independent judgement. Does something look off? Please file an [issue](https://github.com/rmcgibbo/r-rmcgibbo/issues) or reach out on IRC.

06kellyjac commented:
I don't have access to my aarch64 machine atm, if someone could get me some logs that'd be great :sweat_smile:

SuperSandro2000 commented on pkgs/development/tools/just/default.nix:
```diff
@@ -38,22 +38,25 @@ rustPlatform.buildRustPackage rec {
-  # Skip "choose" when running "cargo test", since this test case needs "fzf".
-  checkFlags = [ "--skip=choose" "--skip=edit" ];
+  checkFlags = [
+    "--skip=choose" # test case needs "cat" and "vim"
+    "--skip=edit" # test case needs "fzf"
```
Can we add fzf, coreutils and vim to checkInputs?

06kellyjac replied:
yeah I can try add them in. Is there a concern that it'd bloat the closure size or is it fine as they're added through checkInputs?

SuperSandro2000 replied:
checkInputs shouldn't bloat the closure size. Do we need to patch paths in just to make fzf work?

06kellyjac replied:
So `Command` isn't able to run `vim` even though I've verified it's in the PATH. I'm guessing it falls back to use `/usr/bin/env` or `which` to find where `vim` is (which doesn't exist in the sandbox).
This isn't an issue when using `just` outside of the sandbox. It cant be resolved by patching the test and I don't really want to be patching the source code just for the check phase with: `"s@.unwrap_or_else(|| \"vim\".into());@.unwrap_or_else(|| \"${coreutils}/bin/cat\".into());@g" (patching it for `${vim}/bin/vim` just hangs)`, it's probably best to just skip `edit` with more detailed explanation

https://github.com/casey/just/blob/0a9ffd7a7b1e423e8ae5cee08427a416c957644f/src/config.rs#L686-L691

It's a similar problem for the chooser not being able to find `fzf`. It can be resolved with `export JUST_CHOOSER="${coreutils}/bin/cat"` as there's no test that forces the run of the fallback value.

(As a note if I feed it `export JUST_CHOOSER="${fzf}/bin/fzf"` it doesn't like it and fails with error code 2 `stderr: inappropriate ioctl for device`, probably also due to the sandbox too?)

lukegb commented:
> I don't have access to my aarch64 machine atm, if someone could get me some logs that'd be great 😅

The logs are in the link in the nixpkgs-review comment: https://gist.github.com/r-rmcgibbo/c4abb4cd1293fe0dea89524895c4aefb

```
Executing cargoBuildHook
++ env CC_aarch64-unknown-linux-gnu=/nix/store/vqrgfhnzspbhraalbmwfj6z6zn58d1vw-gcc-wrapper-9.3.0/bin/cc CXX_aarch64-unknown-linux-gnu=/nix/store/vqrgfhnzspbhraalbmwfj6z6zn58d1vw-gcc-wrapper-9.3.0/bin/c++ CC_aarch64-unknown-linux-gnu=/nix/store/vqrgfhnzspbhraalbmwfj6z6zn58d1vw-gcc-wrapper-9.3.0/bin/cc CXX_aarch64-unknown-linux-gnu=/nix/store/vqrgfhnzspbhraalbmwfj6z6zn58d1vw-gcc-wrapper-9.3.0/bin/c++ cargo build -j 2 --target aarch64-unknown-linux-gnu --frozen --release
   Compiling libc v0.2.95
   Compiling cfg-if v1.0.0
   Compiling proc-macro2 v1.0.27
error: linker `aarch64-linux-gnu-gcc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: could not compile `libc`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
error: linker `aarch64-linux-gnu-gcc` not found
  |
  = note: No such file or directory (os error 2)

error: aborting due to previous error

error: build failed
```

06kellyjac committed:
just: 0.9.1 -> 0.9.4

Skip run_shebang test as it randomly fails
Add changelog
Add myself as a maintainer

06kellyjac commented:
@SuperSandro2000 should this have been merged before resolving the aarch64 build issue? As a note I'm not sure what the recommended way of fixing it would be

SuperSandro2000 commented:
Oh, yeah you are probably right. Just assumed it was broken before which it wasn't. We should probably replace ``aarch64-linux-gnu-gcc`` with the right gcc.

06kellyjac commented:
I'm a bit confused why it needs the linker `aarch64-linux-gnu-gcc` when the target is `aarch64-unknown-linux-gnu`

I'm wondering if it's an issue caused the version update to the `libc` crate in just, or if this an issue across all rust aarch64 builds with rust ?

SuperSandro2000 commented:
> I'm a bit confused why it needs the linker `aarch64-linux-gnu-gcc` when the target is `aarch64-unknown-linux-gnu

Yeah, I think normally this is only required for cross compiling.

> I'm wondering if it's an issue caused the version update to the libc crate in just, or if this an issue across all rust aarch64 builds with rust ?

Can't tell you.
